packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts

```diff
@@ -138,24 +145,24 @@ export type TimePickerProps = Omit<
     /**
      * Callback for when a time selection is made.
      */
-    onTimeSelect?: (event: TimeSelectionEvents, data: TimeSelectionData) => void;
+    onTimeChange?: (event: TimeSelectionEvents, data: TimeSelectionData) => void;
 
     /**
-     * Custom the date strings displayed in dropdown options.
+     * Customizes the formatting of date strings displayed in dropdown options.
      */
     formatDateToTimeString?: (date: Date) => string;
 
     /**
-     * Custom validation for the input time string from user in freeform TimePicker.
+     * In the freeform TimePicker, customizes the parsing from the input time string into a Date and provides custom validation.
      */
-    validateFreeFormTime?: (time: string | undefined) => TimeStringValidationResult;
+    formatTimeStringToDate?: (time: string | undefined) => TimeStringValidationResult;
```

Comment:
```suggestion
    formatTimeStringToDate?: (time: string) => Date;
```

- Can we remove `undefined`?
- Can we try/catch from `formatTimeStringToDate` instead?

Reply:
Chat in a call and we're going with keeping the `TimeStringValidationResult` returned type and rename error to errorType